File: disturbances/src/test/java/md/vnastasi/cloud/util/JsonUtils.java
package md.vnastasi.cloud.util;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.lang.NonNull;

import java.io.IOException;
import java.util.Objects;
import java.util.Scanner;

public final class JsonUtils {

    @NonNull
    public static String readString(@NonNull String fileName) {
        var inputStream = JsonUtils.class.getClassLoader().getResourceAsStream("json/" + fileName);
        Objects.requireNonNull(inputStream, String.format("Could not read from file </json/%s>", fileName));
        try (var scanner = new Scanner(inputStream).useDelimiter("\\A")) {
            return scanner.next();
        }
    }

    @NonNull
    public static <T> T deserialize(@NonNull String fileName, @NonNull Class<T> targetClass) throws IOException {
        var jsonString = readString(fileName);
        return new ObjectMapper().readValue(jsonString, targetClass);
    }
}

File: timetable/src/test/kotlin/md/vnastasi/cloud/util/JsonUtils.kt
package md.vnastasi.cloud.util

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue

object JsonUtils {

    fun readString(fileName: String): String {
        val inputStream = javaClass.classLoader.getResourceAsStream("json/$fileName")
        checkNotNull(inputStream) { "Could not read from file </json/$fileName>" }
        return inputStream.bufferedReader().use { it.readText() }
    }

    inline fun <reified T> readObject(fileName: String): T {
        val jsonContents = readString(fileName)
        val objectMapper = ObjectMapper()
            .findAndRegisterModules()
            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
        return objectMapper.readValue(jsonContents)
    }
}
